src/vdaf.rs

```diff
@@ -46,6 +51,24 @@ pub enum Share<F> {
     Helper(Key),
 }
 
+impl<F: FieldElement> Share<F> {
+    pub(crate) fn read_leader<R: Read>(length: usize, reader: &mut R) -> Result<Self, VdafError> {
+        let mut data = Vec::with_capacity(length);
+        for _ in 0..length {
+            data.push(F::try_from_reader(reader).map_err(|_| VdafError::Parse)?);
+        }
+        Ok(Share::Leader(data))
+    }
+
+    pub(crate) fn read_helper<R: Read>(suite: Suite, reader: &mut R) -> Result<Self, VdafError> {
+        let mut seed = Key::uninitialized(suite);
+        reader
+            .read_exact(seed.as_mut_slice())
+            .map_err(|_| VdafError::Parse)?;
```

Comment:
This throws away the underlying error. If you don't want to add a variant to `VdafError` to wrap each of `FieldError, `io::Error` and so on, you could put in a variant like:
```
#[error("Unspecified error {0}")]
Unspecified(#[from] Box<dyn Error>)
```

Then we don't get structure, but at least if something ever logs the error, the underlying error's description gets printed.

Reply:
Ugh, I meant to do this but forgot! Done now.